src/main/java/com/commercetools/sync/products/helpers/VariantReferenceResolver.java

```diff
@@ -184,19 +168,11 @@ static boolean isProductReference(@Nonnull final JsonNode referenceValue) {
     }
 
     @Nonnull
-    CompletionStage<Optional<String>> getResolvedIdFromKeyInReference(@Nonnull final JsonNode referenceValue) {
+    CompletionStage<Optional<String>> getProductResolvedIdFromKeyInReference(@Nonnull final JsonNode referenceValue) {
         final JsonNode idField = referenceValue.get(REFERENCE_ID_FIELD);
         return idField != null
```

Comment:
Really good question. Even though, implementation wise, It would still work without this check. Because eventually, if it doesn't find "null" in the cache, will try to fetch it, and won't find it, so will still return empty optional.

However, having the check as you suggested is better as it avoids having to fetch at all. - Kind of short-circuiting.

Reply:
564c53c